ln-simln-jamming/src/attacks/slow_jam.rs

```diff
@@ -0,0 +1,329 @@
+        let hops = &[self.target_pubkey, self.attacker_receiver.1];
+        let target_channel = (self.target_pubkey, self.channel_to_jam.1);
+
+        let attacker_node_sender = attacker_nodes.get(&self.attacker_sender.0).ok_or(format!(
+            "node {} not found in attacker nodes list",
+            self.attacker_sender.0
+        ))?;
+
+        let build_rep_params = BuildReputationParams {
+            attacker_node: Arc::clone(attacker_node_sender),
+            hops,
+            network_graph: &self.network_graph,
+            // Build reputation for this amount which is slightly above the liquidity available in
+            // the protected bucket of the channel we are trying to jam.
+            htlcs: vec![900_000_000; 1],
+            target_channel,
+            reputation_monitor: Arc::clone(&self.reputation_monitor),
+            payment_hash: PaymentHash(rand::random()),
+            reputation_params: self.reputation_params,
+            clock: Arc::clone(&self.clock),
+            shutdown_listener: trigger().1,
+        };
+        let fees_paid = build_reputation(build_rep_params).await?;
+        Ok(fees_paid)
+    }
+
+    async fn slow_jam_channel(
+        &self,
+        attacker_nodes: &HashMap<String, Arc<Mutex<SimNode<SimGraph, SimulationClock>>>>,
+    ) -> Result<(), BoxError> {
+        // At this point we should:
+        // - Have general and congestion resources jammed.
+        // - Already have built reputation and have access to protected resources.
+        let attacker_node_sender = attacker_nodes.get(&self.attacker_sender.0).ok_or(format!(
+            "node {} not found in attacker nodes list",
+            self.attacker_sender.0
+        ))?;
+
+        // Jam the protected resources of our target channel by sending a payment for the entire
+        // amount of the protected bucket. This will be accepted over the incoming channel because
+        // we have set a channel such that the general slots will accomodate this value.
+        let hops_1 = vec![
+            self.channel_to_jam.0,
+            self.target_pubkey,
+            self.attacker_receiver.1,
+        ];
+        let route = build_custom_route(
+            &self.attacker_sender.1,
+            798_799_000,
+            &hops_1,
+            &self.network_graph,
+        )
+        .map_err(|e| e.err)?;
+
+        // Here attacker receiver must have enough reputation to jam the entire protected bucket.
+        // So that a subsequent honest payment going to an honest receiver can't use the channel
+        // even if it has reputation because all protected resources are taken.
+        let payment_hash = PaymentHash(rand::random());
+        if let Err(e) = attacker_node_sender
+            .lock()
+            .await
+            .send_to_route(route, payment_hash, None)
+            .await
+        {
+            return Err(e.to_string().into());
+        }
+        self.jamming_payments.lock().await.insert(payment_hash);
```

Comment:
The timing of this seems unlikely, but I think it's _possible_ that we get a race where:
* We dispatch the payment
* `intercept_attacker_receive` gets the `jamming_payments` lock
* Payment is not held because not yet added
* We get the `jamming_payments` lock and add it

Can be addressed by just moving this addition up to above the payment dispatch (and can be deleted if we fail to send).